prov/util/src/util_buf.c

```diff
@@ -94,15 +99,22 @@ static int ofi_bufpool_region_alloc(struct ofi_bufpool_region *buf_region)
 		pool->attr.alignment = ofi_get_aligned_size(pool->attr.alignment, page_size);
 	}
 
-	return ofi_memalign((void **) &buf_region->alloc_region,
+	ret = ofi_memalign((void **) &buf_region->alloc_region,
 				roundup_power_of_two(pool->attr.alignment),
 				pool->alloc_size);
+	if (!ret && host_rgister)
+		ofi_hmem_host_register(buf_region->alloc_region, pool->alloc_size);
+	return ret;
 }
 
 static void ofi_bufpool_region_free(struct ofi_bufpool_region *buf_region)
 {
 	int ret;
 	struct ofi_bufpool *pool = buf_region->pool;
+	bool host_register = pool->attr.flags & OFI_BUFPOOL_HMEM_COPY;
+
+	if (host_register)
```

Comment:
at least your spelling is consistent

Reply:
:) that's why the wrongs one was not caught at compile time